code/modules/station_goals/shield.dm

```diff
@@ -0,0 +1,140 @@
+			continue
+		coverage |= view(A.kill_range,A)
+	if(coverage.len >= 400)
+		return TRUE
+	return FALSE
+
+
+/obj/item/weapon/circuitboard/machine/computer/sat_control
+	name = "circuit board (Satellite Network Control)"
+	build_path = /obj/machinery/computer/sat_control
+	origin_tech = "engineering=3"
+
+/obj/machinery/computer/sat_control
+	name = "Satellite control"
+	desc = "Used to control the satellite network."
+	circuit = /obj/item/weapon/circuitboard/machine/computer/sat_control
+	var/notice
+
+/obj/machinery/computer/sat_control/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, \
+										datum/tgui/master_ui = null, datum/ui_state/state = physical_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "sat_control", name, 400, 305, master_ui, state)
+		ui.open()
+
+/obj/machinery/computer/sat_control/ui_act(action, params)
+	if(..())
+		return
+	switch(action)
+		if("toggle")
+			toggle(params["id"])
+			. = TRUE
+
+/obj/machinery/computer/sat_control/proc/toggle(id)
+	for(var/obj/machinery/satellite/S in machines)
+		if(S.id == id && S.z == z)
+			S.toggle()
+	return
+
+/obj/machinery/computer/sat_control/ui_data()
+	var/list/data = list()
+	
+	data["satellites"] = list()
+	for(var/obj/machinery/satellite/S in machines)
+		data["satellites"] += list(list(
+			"id" = S.id,
+			"active" = S.active,
+			"mode" = S.mode
+		))
+	data["notice"] = notice
+	return data
+
+
+/obj/machinery/satellite
+	name = "Defunct Satellite"
+	icon = 'icons/obj/machines/satellite.dmi'
+	icon_state = "sat_inactive"
+	var/mode = "NTPROBEV0.8"
+	var/active = FALSE
+	density = 1
+	var/static/gid = 0
+	var/id = 0
+
+/obj/machinery/satellite/New()
+	..()
+	id = gid++
+
+/obj/machinery/satellite/interact(mob/user)
+	user << "You [active ? "deactivate": "activate"] the [src]"
+	toggle()
+
+/obj/machinery/satellite/proc/toggle()
+	active = !active
+	if(active)
+		animate(src, pixel_y = 2, time = 10, loop = -1)
+		anchored = 1
+	else
+		animate(src, pixel_y = 0, time = 10)
+		anchored = 0
+	update_icon()
+
+/obj/machinery/satellite/update_icon()
+	icon_state = active ? "sat_active" : "sat_inactive"
+
+/obj/machinery/satellite/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/device/multitool))
+		user << "// NTSAT-[id] // Mode : [active ? "PRIMARY" : "STANDBY"] //[emagged ? "DEBUG_MODE //" : ""]"
+	else
+		..()
+
+/obj/machinery/satellite/meteor_shield
+	name = "Meteor Shield Satellite"
+	mode = "M-SHIELD"
+	var/kill_range = 10
+
+/obj/machinery/satellite/meteor_shield/process()
+	if(!active)
+		return
+	for(var/obj/effect/meteor/M in meteor_list)
+		if(M.z != z)
+			continue
+		if(get_dist(M,src) > kill_range)
+			return
+		if(!emagged)
+			qdel(M)
+
+/obj/machinery/satellite/meteor_shield/emag_act()
+	if(!emagged)
+		emagged = 1
+		var/datum/round_event_control/E = locate(/datum/round_event_control/meteor_wave) in SSevent.control
+		if(E)
+			E.weight *= 2
```

Comment:
There needs to be a way to undo this or something. Like make this only occur when the sat is emagged and the network is on.
Otherwise you could just order a bunch of sats and emag them and make the station be bombarded by meteors. no need to set the sats up or anything.